engines/csvengine.py

```diff
@@ -53,27 +55,23 @@ def create_db(self):
     def create_table(self):
         """Create the table by creating an empty csv file"""
         self.auto_column_number = 1
-        self.output_file = open(self.table_name(), "w")
-        self.output_file.write(
-            ','.join(['"%s"' % c for c in self.table.get_insert_columns(join=False,create=True)]))
+        self.file = open(self.table_name(),'w')
+        self.output_file = csv.writer(self.file, dialect='excel', lineterminator='\n', quoting=csv.QUOTE_MINIMAL)
+        self.output_file.writerow(self.table.get_insert_columns(join=False,create=True))
+        self.table_names.append((self.file, self.table_name()))
 
     def disconnect(self):
         """Close the last file in the dataset"""
-        try:
-            self.output_file.close()
-        except:
-            # when disconnect is called by app.connect_wizard.ConfirmPage to
-            # confirm the connection, output_file doesn't exist yet, this is
-            # fine so just pass
-            pass
+        for output_file_i, file_name in self.table_names:
+            output_file_i.close()
 
     def execute(self, statement, commit=True):
         """Write a line to the output file"""
-        self.output_file.write('\n' + statement)
+        self.output_file.writerows(statement)
 
     def format_insert_value(self, value, datatype):
         """Formats a value for an insert statement"""
-        v = Engine.format_insert_value(self, value, datatype, escape=False)
+        v = Engine.format_insert_value(self, value, datatype, escape=False, processed=True)
```

Comment:
If I understand things correctly, this is causing the values to be quoted if they are strings. Is that right? That seems like something that `csv.write` should be able to handle using `quoting=csv.QUOTE_NONNUMERIC`. I'm sure it's more complicated than I'm seeing, so could you walk me through the issue?